Userland/Libraries/LibRegex/RegexMatch.h

```diff
@@ -88,9 +89,19 @@ class RegexStringView {
             return new_views;
         }
 
-        // FIXME: line splitting for Utf32View needed
         Vector<RegexStringView> views;
-        views.append(m_u32view.value());
+        auto view = u32view();
+        u32 newline = '\n';
+        for (; !view.is_empty();) {
```

Comment:
Is there a reason for using `for(; foo();)` over `while(foo())`?

Reply:
Not really, just something my brain came up with on the spot
I don't mind either